### PR TITLE
[CI] New unit tests pipeline with Github Actions

### PR DESCRIPTION
**Pull request by odesenfans**
Added a new CI pipeline to test the codebase with the Go and JS
P2P daemons. This pipeline tests the codebase against different
versions of Python, the Go daemon v0.2.0 and the JS daemon v0.10.2.

### CODE REVIEW

**Comment by odesenfans** [comment]
Here's a first PR for the CI. This focuses on running the unit tests against both daemons. Notes:
* I had to pick v0.2.0 for the Go daemon, v0.3.0 is acting up and master as well. Given that it's unmaintained, I say good enough.
* I'm planning to add a typing check, but there are some fixes to make so that's for another PR.
* I'll use another pipeline for the releases. What I'd like to have is a pipeline triggered on each new tag on Github that makes a few checks (ex: does the version of the package match the version of the tag) and then pushes to pypy.

**Comment by odesenfans** [comment]
I think we will need to merge this one, or add a dummy pipeline in master to trigger the actions on this repository. In the meantime, you can check the results here: https://github.com/odesenfans/py-libp2p-daemon-bindings/actions/runs/1797548118

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Unit tests
+
+on:
+  push:
+    branches:
+      - master
+      - od-ci-github-actions
```
Comment (mhchia) [inline .github/workflows/unit-tests.yml:10]: Is it a leftover?

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    branches:
+      - master
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # Tests disabled on Python 3.10 because of an issue with pytest
```
Comment (mhchia) [inline .github/workflows/unit-tests.yml:21]: Can I know what's the issue, just curious?

File: .github/workflows/unit-tests.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
```
Comment (mhchia) [inline .github/workflows/unit-tests.yml:1]:
```suggestion
# This workflow will install Python dependencies, run tests with a variety of Python versions
```

**Review by mhchia — APPROVED** [review]
Nice work! Left some comments. Feel free to merge whenever you think it's ready.

**Comment by mhchia** [comment]
> I had to pick v0.2.0 for the Go daemon, v0.3.0 is acting up and master as well. Given that it's unmaintained, I say good enough.
I'm planning to add a typing check, but there are some fixes to make so that's for another PR.
I'll use another pipeline for the releases. What I'd like to have is a pipeline triggered on each new tag on Github that makes a few checks (ex: does the version of the package match the version of the tag) and then pushes to pypy.

All sounds good to me!

**Comment by odesenfans** [comment]
You can check the issue on this build I just triggered: https://github.com/odesenfans/py-libp2p-daemon-bindings/runs/5083468843?check_suite_focus=true

To be resolved in a later PR.

**Pull request merged**